posthog/api/dashboard.py

```diff
@@ -96,16 +98,18 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Dashboard:
                         "dashboard": dashboard.pk,
                         "last_refresh": now(),
                     }
-                    insight_serializer = InsightSerializer(
-                        data={
-                            **InsightSerializer(dashboard_item, context=self.context,).data,
-                            **override_dashboard_item_data,
-                        },
-                        context=self.context,
-                    )
+                    new_data = {
+                        **InsightSerializer(dashboard_item, context=self.context,).data,
+                        **override_dashboard_item_data,
+                    }
+                    new_tags = new_data.pop("tags", None)
+                    insight_serializer = InsightSerializer(data=new_data, context=self.context,)
                     insight_serializer.is_valid()
                     insight_serializer.save()
 
+                    # Create new insight's tags separately. Force create tags on dashboard duplication.
+                    self._attempt_set_tags(new_tags, insight_serializer.instance, True)
```

Comment:
Super nit: i'd use the named `force_create` arg, so it's a bit clearer what the `True` is.

Reply:
Good catch on the duplication case!